PF/PF-Assignment2/ShoppingCart.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Code by __E18/1025__
+ * Shopping Cart for item 
+ */
+
+public class ShoppingCart {
+	private Item firstItem ;
+	private Item lastItem;
+	private double total;
+	
+	ShoppingCart(){
+		this.firstItem = null;
+		this.lastItem = null;
+		this.total = 0.0;
+	}
+	
+	/*
+	 * Enum for Promotion code 
+	 */
+	enum PromotionEnum{
+		SkyBag("sky10", "15/07/2018", "20/07/2018"),
+		Algo_Book("algo10", "15/07/2018", "20/07/2018"),
+		Laptop("laptop10", "15/07/2018", "20/07/2018"),
+		Samsung("samsung10", "15/07/2018", "20/07/2018");
+		
+		private String code, startDate, endDate;
+		
+		/*
+		 * @param   nothing
+		 * @return  code 
+		 */
+		public String getCode() {
+			return this.code;
+		}
+		
+		/*
+		 * @param   nothing
+		 * @return  start date 
+		 */
+		public String getStartDate() {
+			return this.startDate;
+		}
+		
+		/*
+		 * @param   nothing
+		 * @return  end date
+		 */
+		public String getEndDate() {
+			return this.endDate;
+		}
+		
+		private PromotionEnum(String code, String startDate, String endDate) {
+			this.code = code;
+			this.startDate = startDate;
+			this.endDate = endDate;
+		}
+	}
+	
+	static class Item{
+		private long itemID;
+		private String itemName;
+		private double price;
+		private long quantity;
+		private Item next;
+		
+		Item(long itemID, String itemName, double price, long quantity){
+			this.itemID = itemID;
+			this.itemName = itemName;
+			this.price = price;
+			this.quantity = quantity;
+			this.next = null;
+		}
+	}
+	
+	/*
+	 * @param   item id (int)
+	 * @return  item price
+	 */
+	public double getPrice(long itemID) {
+		Item item;
+		item = this.searchItem(itemID);
+		return item.price;
+	}
+	
+	/*
+	 * @param   item id (int), price of item (double)
+	 * @return  nothing
+	 * Update Item price after discount
+	 */
+	public void setPrice(long itemID, double price) {
+		Item item;
+		item = this.searchItem(itemID);
+		item.price = price;
+	}
+	
+	/*
+	 * @param   nothing
+	 * @return  true if Shopping Cart is Empty otherwise false
+	 */
+	public boolean isEmpty() {
+		if (this.firstItem == null)
+			return true;
+		return false;
+	}
+	
+	/*
+	 * @param   item name (string), item price (double), item quantity(long)
+	 * @return  nothing
+	 * Add item into the Shopping Cart list
+	 */
+	public void add(long itemID, String itemName, double price, long quantity) {
+		Item newItem = new Item(itemID, itemName, price, quantity);
+		total += price * (double)quantity;
```

Comment:
setTotal or calculateTotal will do the total calculation i.e. taking itemLines from cart and sum of line totals will give the net total.